src/types/complex.rs

```diff
@@ -289,24 +281,24 @@ impl<'py> PyComplexMethods<'py> for Bound<'py, PyComplex> {
 
     #[cfg(not(any(Py_LIMITED_API, PyPy, GraalPy)))]
     fn abs(&self) -> c_double {
-        unsafe {
-            let val = (*self.as_ptr().cast::<ffi::PyComplexObject>()).cval;
-            ffi::_Py_c_abs(val)
-        }
+        self.as_borrowed()
+            .getattr("__abs__")
+            .expect("Cannot find `__abs__` method.")
+            .call0()
+            .expect("Failed to call `__abs__` method.")
+            .downcast_into()
+            .expect("Failed to downcast to complex number.")
+            .extract()
+            .expect("Failed to extract to c double.")
```

Comment:
Similar comment that I this should go through `PyNumber_Absolute` and we could add `PyAnyMethods::abs`.